// grpc-js-reflection-client/dist/Proto/v1.js
"use strict";
var __classPrivateFieldGet = (this && this.__classPrivateFieldGet) || function (receiver, state, kind, f) {
    if (kind === "a" && !f) throw new TypeError("Private accessor was defined without a getter");
    if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver)) throw new TypeError("Cannot read private member from an object whose class did not declare it");
    return kind === "m" ? f : kind === "a" ? f.call(receiver) : f ? f.value : state.get(receiver);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.grpc = void 0;
const pb_1 = require("google-protobuf");
const grpc_1 = require("@grpc/grpc-js");
var grpc;
(function (grpc) {
    var reflection;
    (function (reflection) {
        var v1;
        (function (v1) {
            var _ServerReflectionRequest_one_of_decls, _ExtensionRequest_one_of_decls, _ServerReflectionResponse_one_of_decls, _FileDescriptorResponse_one_of_decls, _ExtensionNumberResponse_one_of_decls, _ListServiceResponse_one_of_decls, _ServiceResponse_one_of_decls, _ErrorResponse_one_of_decls;
            class ServerReflectionRequest extends pb_1.Message {
                constructor(data) {
                    super();
                    _ServerReflectionRequest_one_of_decls.set(this, [[3, 4, 5, 6, 7]]);
                    pb_1.Message.initialize(this, Array.isArray(data) ? data : [], 0, -1, [], __classPrivateFieldGet(this, _ServerReflectionRequest_one_of_decls, "f"));
                    if (!Array.isArray(data) && typeof data == "object") {
                        if ("host" in data && data.host != undefined) {
                            this.host = data.host;
                        }
                        if ("fileByFilename" in data && data.fileByFilename != undefined) {
                            this.fileByFilename = data.fileByFilename;
                        }
                        if ("fileContainingSymbol" in data && data.fileContainingSymbol != undefined) {
                            this.fileContainingSymbol = data.fileContainingSymbol;
                        }
                        if ("fileContainingExtension" in data && data.fileContainingExtension != undefined) {
                            this.fileContainingExtension = data.fileContainingExtension;
                        }
                        if ("allExtensionNumbersOfType" in data && data.allExtensionNumbersOfType != undefined) {
                            this.allExtensionNumbersOfType = data.allExtensionNumbersOfType;
                        }
                        if ("listServices" in data && data.listServices != undefined) {
                            this.listServices = data.listServices;
                        }
                    }
                }
                get host() {
                    return pb_1.Message.getFieldWithDefault(this, 1, "");
                }
                set host(value) {
                    pb_1.Message.setField(this, 1, value);
                }
                get fileByFilename() {
                    return pb_1.Message.getFieldWithDefault(this, 3, "");
                }
                set fileByFilename(value) {
                    pb_1.Message.setOneofField(this, 3, __classPrivateFieldGet(this, _ServerReflectionRequest_one_of_decls, "f")[0], value);
                }
                get hasFileByFilename() {
                    return pb_1.Message.getField(this, 3) != null;
                }
                get fileContainingSymbol() {
                    return pb_1.Message.getFieldWithDefault(this, 4, "");
                }
                set fileContainingSymbol(value) {
                    pb_1.Message.setOneofField(this, 4, __classPrivateFieldGet(this, _ServerReflectionRequest_one_of_decls, "f")[0], value);
                }
                get hasFileContainingSymbol() {
                    return pb_1.Message.getField(this, 4) != null;
                }
                get fileContainingExtension() {
                    return pb_1.Message.getWrapperField(this, ExtensionRequest, 5);
                }
                set fileContainingExtension(value) {
                    pb_1.Message.setOneofWrapperField(this, 5, __classPrivateFieldGet(this, _ServerReflectionRequest_one_of_decls, "f")[0], value);
                }
                get hasFileContainingExtension() {
                    return pb_1.Message.getField(this, 5) != null;
                }
                get allExtensionNumbersOfType() {
                    return pb_1.Message.getFieldWithDefault(this, 6, "");
                }
                set allExtensionNumbersOfType(value) {
                    pb_1.Message.setOneofField(this, 6, __classPrivateFieldGet(this, _ServerReflectionRequest_one_of_decls, "f")[0], value);
                }
                get hasAllExtensionNumbersOfType() {
                    return pb_1.Message.getField(this, 6) != null;
                }
                get listServices() {
                    return pb_1.Message.getFieldWithDefault(this, 7, "");
                }
                set listServices(value) {
                    pb_1.Message.setOneofField(this, 7, __classPrivateFieldGet(this, _ServerReflectionRequest_one_of_decls, "f")[0], value);
                }
                get hasListServices() {
                    return pb_1.Message.getField(this, 7) != null;
                }
                get message_request() {
                    const cases = {
                        0: "none",
                        3: "fileByFilename",
                        4: "fileContainingSymbol",
                        5: "fileContainingExtension",
                        6: "allExtensionNumbersOfType",
                        7: "listServices"
                    };
                    return cases[pb_1.Message.computeOneofCase(this, [3, 4, 5, 6, 7])];
                }
                static fromObject(data) {
                    const message = new ServerReflectionRequest({});
                    if (data.host != null) {
                        message.host = data.host;
                    }
                    if (data.fileByFilename != null) {
                        message.fileByFilename = data.fileByFilename;
                    }
                    if (data.fileContainingSymbol != null) {
                        message.fileContainingSymbol = data.fileContainingSymbol;
                    }
                    if (data.fileContainingExtension != null) {
                        message.fileContainingExtension = ExtensionRequest.fromObject(data.fileContainingExtension);
                    }
                    if (data.allExtensionNumbersOfType != null) {
                        message.allExtensionNumbersOfType = data.allExtensionNumbersOfType;
                    }
                    if (data.listServices != null) {
                        message.listServices = data.listServices;
                    }
                    return message;
                }
                toObject() {
                    const data = {};
                    if (this.host != null) {
                        data.host = this.host;
                    }
                    if (this.fileByFilename != null) {
                        data.fileByFilename = this.fileByFilename;
                    }
                    if (this.fileContainingSymbol != null) {
                        data.fileContainingSymbol = this.fileContainingSymbol;
                    }
                    if (this.fileContainingExtension != null) {
                        data.fileContainingExtension = this.fileContainingExtension.toObject();
                    }
                    if (this.allExtensionNumbersOfType != null) {
                        data.allExtensionNumbersOfType = this.allExtensionNumbersOfType;
                    }
                    if (this.listServices != null) {
                        data.listServices = this.listServices;
                    }
                    return data;
                }
                serialize(w) {
                    const writer = w || new pb_1.BinaryWriter();
                    if (this.host.length)
                        writer.writeString(1, this.host);
                    if (this.hasFileByFilename)
                        writer.writeString(3, this.fileByFilename);
                    if (this.hasFileContainingSymbol)
                        writer.writeString(4, this.fileContainingSymbol);
                    if (this.hasFileContainingExtension)
                        writer.writeMessage(5, this.fileContainingExtension, () => this.fileContainingExtension.serialize(writer));
                    if (this.hasAllExtensionNumbersOfType)
                        writer.writeString(6, this.allExtensionNumbersOfType);
                    if (this.hasListServices)
                        writer.writeString(7, this.listServices);
                    if (!w)
                        return writer.getResultBuffer();
                }
                static deserialize(bytes) {
                    const reader = bytes instanceof pb_1.BinaryReader ? bytes : new pb_1.BinaryReader(bytes), message = new ServerReflectionRequest();
                    while (reader.nextField()) {
                        if (reader.isEndGroup())
                            break;
                        switch (reader.getFieldNumber()) {
                            case 1:
                                message.host = reader.readString();
                                break;
                            case 3:
                                message.fileByFilename = reader.readString();
                                break;
                            case 4:
                                message.fileContainingSymbol = reader.readString();
                                break;
                            case 5:
                                reader.readMessage(message.fileContainingExtension, () => message.fileContainingExtension = ExtensionRequest.deserialize(reader));
                                break;
                            case 6:
                                message.allExtensionNumbersOfType = reader.readString();
                                break;
                            case 7:
                                message.listServices = reader.readString();
                                break;
                            default: reader.skipField();
                        }
                    }
                    return message;
                }
                serializeBinary() {
                    return this.serialize();
                }
                static deserializeBinary(bytes) {
                    return ServerReflectionRequest.deserialize(bytes);
                }
            }
            _ServerReflectionRequest_one_of_decls = new WeakMap();
            v1.ServerReflectionRequest = ServerReflectionRequest;
            class ExtensionRequest extends pb_1.Message {
                constructor(data) {
                    super();
                    _ExtensionRequest_one_of_decls.set(this, []);
                    pb_1.Message.initialize(this, Array.isArray(data) ? data : [], 0, -1, [], __classPrivateFieldGet(this, _ExtensionRequest_one_of_decls, "f"));
                    if (!Array.isArray(data) && typeof data == "object") {
                        if ("containingType" in data && data.containingType != undefined) {
                            this.containingType = data.containingType;
                        }
                        if ("extensionNumber" in data && data.extensionNumber != undefined) {
                            this.extensionNumber = data.extensionNumber;
                        }
                    }
                }
                get containingType() {
                    return pb_1.Message.getFieldWithDefault(this, 1, "");
                }
                set containingType(value) {
                    pb_1.Message.setField(this, 1, value);
                }
                get extensionNumber() {
                    return pb_1.Message.getFieldWithDefault(this, 2, 0);
                }
                set extensionNumber(value) {
                    pb_1.Message.setField(this, 2, value);
                }
                static fromObject(data) {
                    const message = new ExtensionRequest({});
                    if (data.containingType != null) {
                        message.containingType = data.containingType;
                    }
                    if (data.extensionNumber != null) {
                        message.extensionNumber = data.extensionNumber;
                    }
                    return message;
                }
                toObject() {
                    const data = {};
                    if (this.containingType != null) {
                        data.containingType = this.containingType;
                    }
                    if (this.extensionNumber != null) {
                        data.extensionNumber = this.extensionNumber;
                    }
                    return data;
                }
                serialize(w) {
                    const writer = w || new pb_1.BinaryWriter();
                    if (this.containingType.length)
                        writer.writeString(1, this.containingType);
                    if (this.extensionNumber != 0)
                        writer.writeInt32(2, this.extensionNumber);
                    if (!w)
                        return writer.getResultBuffer();
                }
                static deserialize(bytes) {
                    const reader = bytes instanceof pb_1.BinaryReader ? bytes : new pb_1.BinaryReader(bytes), message = new ExtensionRequest();
                    while (reader.nextField()) {
                        if (reader.isEndGroup())
                            break;
                        switch (reader.getFieldNumber()) {
                            case 1:
                                message.containingType = reader.readString();
                                break;
                            case 2:
                                message.extensionNumber = reader.readInt32();
                                break;
                            default: reader.skipField();
                        }
                    }
                    return message;
                }
                serializeBinary() {
                    return this.serialize();
                }
                static deserializeBinary(bytes) {
                    return ExtensionRequest.deserialize(bytes);
                }
            }
            _ExtensionRequest_one_of_decls = new WeakMap();
            v1.ExtensionRequest = ExtensionRequest;
            class ServerReflectionResponse extends pb_1.Message {
                constructor(data) {
                    super();
                    _ServerReflectionResponse_one_of_decls.set(this, [[4, 5, 6, 7]]);
                    pb_1.Message.initialize(this, Array.isArray(data) ? data : [], 0, -1, [], __classPrivateFieldGet(this, _ServerReflectionResponse_one_of_decls, "f"));
                    if (!Array.isArray(data) && typeof data == "object") {
                        if ("validHost" in data && data.validHost != undefined) {
                            this.validHost = data.validHost;
                        }
                        if ("originalRequest" in data && data.originalRequest != undefined) {
                            this.originalRequest = data.originalRequest;
                        }
                        if ("fileDescriptorResponse" in data && data.fileDescriptorResponse != undefined) {
                            this.fileDescriptorResponse = data.fileDescriptorResponse;
                        }
                        if ("allExtensionNumbersResponse" in data && data.allExtensionNumbersResponse != undefined) {
                            this.allExtensionNumbersResponse = data.allExtensionNumbersResponse;
                        }
                        if ("listServicesResponse" in data && data.listServicesResponse != undefined) {
                            this.listServicesResponse = data.listServicesResponse;
                        }
                        if ("errorResponse" in data && data.errorResponse != undefined) {
                            this.errorResponse = data.errorResponse;
                        }
                    }
                }
                get validHost() {
                    return pb_1.Message.getFieldWithDefault(this, 1, "");
                }
                set validHost(value) {
                    pb_1.Message.setField(this, 1, value);
                }
                get originalRequest() {
                    return pb_1.Message.getWrapperField(this, ServerReflectionRequest, 2);
                }
                set originalRequest(value) {
                    pb_1.Message.setWrapperField(this, 2, value);
                }
                get hasOriginalRequest() {
                    return pb_1.Message.getField(this, 2) != null;
                }
                get fileDescriptorResponse() {
                    return pb_1.Message.getWrapperField(this, FileDescriptorResponse, 4);
                }
                set fileDescriptorResponse(value) {
                    pb_1.Message.setOneofWrapperField(this, 4, __classPrivateFieldGet(this, _ServerReflectionResponse_one_of_decls, "f")[0], value);
                }
                get hasFileDescriptorResponse() {
                    return pb_1.Message.getField(this, 4) != null;
                }
                get allExtensionNumbersResponse() {
                    return pb_1.Message.getWrapperField(this, ExtensionNumberResponse, 5);
                }
                set allExtensionNumbersResponse(value) {
                    pb_1.Message.setOneofWrapperField(this, 5, __classPrivateFieldGet(this, _ServerReflectionResponse_one_of_decls, "f")[0], value);
                }
                get hasAllExtensionNumbersResponse() {
                    return pb_1.Message.getField(this, 5) != null;
                }
                get listServicesResponse() {
                    return pb_1.Message.getWrapperField(this, ListServiceResponse, 6);
                }
                set listServicesResponse(value) {
                    pb_1.Message.setOneofWrapperField(this, 6, __classPrivateFieldGet(this, _ServerReflectionResponse_one_of_decls, "f")[0], value);
                }
                get hasListServicesResponse() {
                    return pb_1.Message.getField(this, 6) != null;
                }
                get errorResponse() {
                    return pb_1.Message.getWrapperField(this, ErrorResponse, 7);
                }
                set errorResponse(value) {
                    pb_1.Message.setOneofWrapperField(this, 7, __classPrivateFieldGet(this, _ServerReflectionResponse_one_of_decls, "f")[0], value);
                }
                get hasErrorResponse() {
                    return pb_1.Message.getField(this, 7) != null;
                }
                get message_response() {
                    const cases = {
                        0: "none",
                        4: "fileDescriptorResponse",
                        5: "allExtensionNumbersResponse",
                        6: "listServicesResponse",
                        7: "errorResponse"
                    };
                    return cases[pb_1.Message.computeOneofCase(this, [4, 5, 6, 7])];
                }
                static fromObject(data) {
                    const message = new ServerReflectionResponse({});
                    if (data.validHost != null) {
                        message.validHost = data.validHost;
                    }
                    if (data.originalRequest != null) {
                        message.originalRequest = ServerReflectionRequest.fromObject(data.originalRequest);
                    }
                    if (data.fileDescriptorResponse != null) {
                        message.fileDescriptorResponse = FileDescriptorResponse.fromObject(data.fileDescriptorResponse);
                    }
                    if (data.allExtensionNumbersResponse != null) {
                        message.allExtensionNumbersResponse = ExtensionNumberResponse.fromObject(data.allExtensionNumbersResponse);
                    }
                    if (data.listServicesResponse != null) {
                        message.listServicesResponse = ListServiceResponse.fromObject(data.listServicesResponse);
                    }
                    if (data.errorResponse != null) {
                        message.errorResponse = ErrorResponse.fromObject(data.errorResponse);
                    }
                    return message;
                }
                toObject() {
                    const data = {};
                    if (this.validHost != null) {
                        data.validHost = this.validHost;
                    }
                    if (this.originalRequest != null) {
                        data.originalRequest = this.originalRequest.toObject();
                    }
                    if (this.fileDescriptorResponse != null) {
                        data.fileDescriptorResponse = this.fileDescriptorResponse.toObject();
                    }
                    if (this.allExtensionNumbersResponse != null) {
                        data.allExtensionNumbersResponse = this.allExtensionNumbersResponse.toObject();
                    }
                    if (this.listServicesResponse != null) {
                        data.listServicesResponse = this.listServicesResponse.toObject();
                    }
                    if (this.errorResponse != null) {
                        data.errorResponse = this.errorResponse.toObject();
                    }
                    return data;
                }
                serialize(w) {
                    const writer = w || new pb_1.BinaryWriter();
                    if (this.validHost.length)
                        writer.writeString(1, this.validHost);
                    if (this.hasOriginalRequest)
                        writer.writeMessage(2, this.originalRequest, () => this.originalRequest.serialize(writer));
                    if (this.hasFileDescriptorResponse)
                        writer.writeMessage(4, this.fileDescriptorResponse, () => this.fileDescriptorResponse.serialize(writer));
                    if (this.hasAllExtensionNumbersResponse)
                        writer.writeMessage(5, this.allExtensionNumbersResponse, () => this.allExtensionNumbersResponse.serialize(writer));
                    if (this.hasListServicesResponse)
                        writer.writeMessage(6, this.listServicesResponse, () => this.listServicesResponse.serialize(writer));
                    if (this.hasErrorResponse)
                        writer.writeMessage(7, this.errorResponse, () => this.errorResponse.serialize(writer));
                    if (!w)
                        return writer.getResultBuffer();
                }
                static deserialize(bytes) {
                    const reader = bytes instanceof pb_1.BinaryReader ? bytes : new pb_1.BinaryReader(bytes), message = new ServerReflectionResponse();
                    while (reader.nextField()) {
                        if (reader.isEndGroup())
                            break;
                        switch (reader.getFieldNumber()) {
                            case 1:
                                message.validHost = reader.readString();
                                break;
                            case 2:
                                reader.readMessage(message.originalRequest, () => message.originalRequest = ServerReflectionRequest.deserialize(reader));
                                break;
                            case 4:
                                reader.readMessage(message.fileDescriptorResponse, () => message.fileDescriptorResponse = FileDescriptorResponse.deserialize(reader));
                                break;
                            case 5:
                                reader.readMessage(message.allExtensionNumbersResponse, () => message.allExtensionNumbersResponse = ExtensionNumberResponse.deserialize(reader));
                                break;
                            case 6:
                                reader.readMessage(message.listServicesResponse, () => message.listServicesResponse = ListServiceResponse.deserialize(reader));
                                break;
                            case 7:
                                reader.readMessage(message.errorResponse, () => message.errorResponse = ErrorResponse.deserialize(reader));
                                break;
                            default: reader.skipField();
                        }
                    }
                    return message;
                }
                serializeBinary() {
                    return this.serialize();
                }
                static deserializeBinary(bytes) {
                    return ServerReflectionResponse.deserialize(bytes);
                }
            }
            _ServerReflectionResponse_one_of_decls = new WeakMap();
            v1.ServerReflectionResponse = ServerReflectionResponse;
            class FileDescriptorResponse extends pb_1.Message {
                constructor(data) {
                    super();
                    _FileDescriptorResponse_one_of_decls.set(this, []);
                    pb_1.Message.initialize(this, Array.isArray(data) ? data : [], 0, -1, [1], __classPrivateFieldGet(this, _FileDescriptorResponse_one_of_decls, "f"));
                    if (!Array.isArray(data) && typeof data == "object") {
                        if ("fileDescriptorProto" in data && data.fileDescriptorProto != undefined) {
                            this.fileDescriptorProto = data.fileDescriptorProto;
                        }
                    }
                }
                get fileDescriptorProto() {
                    return pb_1.Message.getFieldWithDefault(this, 1, []);
                }
                set fileDescriptorProto(value) {
                    pb_1.Message.setField(this, 1, value);
                }
                static fromObject(data) {
                    const message = new FileDescriptorResponse({});
                    if (data.fileDescriptorProto != null) {
                        message.fileDescriptorProto = data.fileDescriptorProto;
                    }
                    return message;
                }
                toObject() {
                    const data = {};
                    if (this.fileDescriptorProto != null) {
                        data.fileDescriptorProto = this.fileDescriptorProto;
                    }
                    return data;
                }
                serialize(w) {
                    const writer = w || new pb_1.BinaryWriter();
                    if (this.fileDescriptorProto.length)
                        writer.writeRepeatedBytes(1, this.fileDescriptorProto);
                    if (!w)
                        return writer.getResultBuffer();
                }
                static deserialize(bytes) {
                    const reader = bytes instanceof pb_1.BinaryReader ? bytes : new pb_1.BinaryReader(bytes), message = new FileDescriptorResponse();
                    while (reader.nextField()) {
                        if (reader.isEndGroup())
                            break;
                        switch (reader.getFieldNumber()) {
                            case 1:
                                pb_1.Message.addToRepeatedField(message, 1, reader.readBytes());
                                break;
                            default: reader.skipField();
                        }
                    }
                    return message;
                }
                serializeBinary() {
                    return this.serialize();
                }
                static deserializeBinary(bytes) {
                    return FileDescriptorResponse.deserialize(bytes);
                }
            }
            _FileDescriptorResponse_one_of_decls = new WeakMap();
            v1.FileDescriptorResponse = FileDescriptorResponse;
            class ExtensionNumberResponse extends pb_1.Message {
                constructor(data) {
                    super();
                    _ExtensionNumberResponse_one_of_decls.set(this, []);
                    pb_1.Message.initialize(this, Array.isArray(data) ? data : [], 0, -1, [2], __classPrivateFieldGet(this, _ExtensionNumberResponse_one_of_decls, "f"));
                    if (!Array.isArray(data) && typeof data == "object") {
                        if ("baseTypeName" in data && data.baseTypeName != undefined) {
                            this.baseTypeName = data.baseTypeName;
                        }
                        if ("extensionNumber" in data && data.extensionNumber != undefined) {
                            this.extensionNumber = data.extensionNumber;
                        }
                    }
                }
                get baseTypeName() {
                    return pb_1.Message.getFieldWithDefault(this, 1, "");
                }
                set baseTypeName(value) {
                    pb_1.Message.setField(this, 1, value);
                }
                get extensionNumber() {
                    return pb_1.Message.getFieldWithDefault(this, 2, []);
                }
                set extensionNumber(value) {
                    pb_1.Message.setField(this, 2, value);
                }
                static fromObject(data) {
                    const message = new ExtensionNumberResponse({});
                    if (data.baseTypeName != null) {
                        message.baseTypeName = data.baseTypeName;
                    }
                    if (data.extensionNumber != null) {
                        message.extensionNumber = data.extensionNumber;
                    }
                    return message;
                }
                toObject() {
                    const data = {};
                    if (this.baseTypeName != null) {
                        data.baseTypeName = this.baseTypeName;
                    }
                    if (this.extensionNumber != null) {
                        data.extensionNumber = this.extensionNumber;
                    }
                    return data;
                }
                serialize(w) {
                    const writer = w || new pb_1.BinaryWriter();
                    if (this.baseTypeName.length)
                        writer.writeString(1, this.baseTypeName);
                    if (this.extensionNumber.length)
                        writer.writePackedInt32(2, this.extensionNumber);
                    if (!w)
                        return writer.getResultBuffer();
                }
                static deserialize(bytes) {
                    const reader = bytes instanceof pb_1.BinaryReader ? bytes : new pb_1.BinaryReader(bytes), message = new ExtensionNumberResponse();
                    while (reader.nextField()) {
                        if (reader.isEndGroup())
                            break;
                        switch (reader.getFieldNumber()) {
                            case 1:
                                message.baseTypeName = reader.readString();
                                break;
                            case 2:
                                message.extensionNumber = reader.readPackedInt32();
                                break;
                            default: reader.skipField();
                        }
                    }
                    return message;
                }
                serializeBinary() {
                    return this.serialize();
                }
                static deserializeBinary(bytes) {
                    return ExtensionNumberResponse.deserialize(bytes);
                }
            }
            _ExtensionNumberResponse_one_of_decls = new WeakMap();
            v1.ExtensionNumberResponse = ExtensionNumberResponse;
            class ListServiceResponse extends pb_1.Message {
                constructor(data) {
                    super();
                    _ListServiceResponse_one_of_decls.set(this, []);
                    pb_1.Message.initialize(this, Array.isArray(data) ? data : [], 0, -1, [1], __classPrivateFieldGet(this, _ListServiceResponse_one_of_decls, "f"));
                    if (!Array.isArray(data) && typeof data == "object") {
                        if ("service" in data && data.service != undefined) {
                            this.service = data.service;
                        }
                    }
                }
                get service() {
                    return pb_1.Message.getRepeatedWrapperField(this, ServiceResponse, 1);
                }
                set service(value) {
                    pb_1.Message.setRepeatedWrapperField(this, 1, value);
                }
                static fromObject(data) {
                    const message = new ListServiceResponse({});
                    if (data.service != null) {
                        message.service = data.service.map(item => ServiceResponse.fromObject(item));
                    }
                    return message;
                }
                toObject() {
                    const data = {};
                    if (this.service != null) {
                        data.service = this.service.map((item) => item.toObject());
                    }
                    return data;
                }
                serialize(w) {
                    const writer = w || new pb_1.BinaryWriter();
                    if (this.service.length)
                        writer.writeRepeatedMessage(1, this.service, (item) => item.serialize(writer));
                    if (!w)
                        return writer.getResultBuffer();
                }
                static deserialize(bytes) {
                    const reader = bytes instanceof pb_1.BinaryReader ? bytes : new pb_1.BinaryReader(bytes), message = new ListServiceResponse();
                    while (reader.nextField()) {
                        if (reader.isEndGroup())
                            break;
                        switch (reader.getFieldNumber()) {
                            case 1:
                                reader.readMessage(message.service, () => pb_1.Message.addToRepeatedWrapperField(message, 1, ServiceResponse.deserialize(reader), ServiceResponse));
                                break;
                            default: reader.skipField();
                        }
                    }
                    return message;
                }
                serializeBinary() {
                    return this.serialize();
                }
                static deserializeBinary(bytes) {
                    return ListServiceResponse.deserialize(bytes);
                }
            }
            _ListServiceResponse_one_of_decls = new WeakMap();
            v1.ListServiceResponse = ListServiceResponse;
            class ServiceResponse extends pb_1.Message {
                constructor(data) {
                    super();
                    _ServiceResponse_one_of_decls.set(this, []);
                    pb_1.Message.initialize(this, Array.isArray(data) ? data : [], 0, -1, [], __classPrivateFieldGet(this, _ServiceResponse_one_of_decls, "f"));
                    if (!Array.isArray(data) && typeof data == "object") {
                        if ("name" in data && data.name != undefined) {
                            this.name = data.name;
                        }
                    }
                }
                get name() {
                    return pb_1.Message.getFieldWithDefault(this, 1, "");
                }
                set name(value) {
                    pb_1.Message.setField(this, 1, value);
                }
                static fromObject(data) {
                    const message = new ServiceResponse({});
                    if (data.name != null) {
                        message.name = data.name;
                    }
                    return message;
                }
                toObject() {
                    const data = {};
                    if (this.name != null) {
                        data.name = this.name;
                    }
                    return data;
                }
                serialize(w) {
                    const writer = w || new pb_1.BinaryWriter();
                    if (this.name.length)
                        writer.writeString(1, this.name);
                    if (!w)
                        return writer.getResultBuffer();
                }
                static deserialize(bytes) {
                    const reader = bytes instanceof pb_1.BinaryReader ? bytes : new pb_1.BinaryReader(bytes), message = new ServiceResponse();
                    while (reader.nextField()) {
                        if (reader.isEndGroup())
                            break;
                        switch (reader.getFieldNumber()) {
                            case 1:
                                message.name = reader.readString();
                                break;
                            default: reader.skipField();
                        }
                    }
                    return message;
                }
                serializeBinary() {
                    return this.serialize();
                }
                static deserializeBinary(bytes) {
                    return ServiceResponse.deserialize(bytes);
                }
            }
            _ServiceResponse_one_of_decls = new WeakMap();
            v1.ServiceResponse = ServiceResponse;
            class ErrorResponse extends pb_1.Message {
                constructor(data) {
                    super();
                    _ErrorResponse_one_of_decls.set(this, []);
                    pb_1.Message.initialize(this, Array.isArray(data) ? data : [], 0, -1, [], __classPrivateFieldGet(this, _ErrorResponse_one_of_decls, "f"));
                    if (!Array.isArray(data) && typeof data == "object") {
                        if ("errorCode" in data && data.errorCode != undefined) {
                            this.errorCode = data.errorCode;
                        }
                        if ("errorMessage" in data && data.errorMessage != undefined) {
                            this.errorMessage = data.errorMessage;
                        }
                    }
                }
                get errorCode() {
                    return pb_1.Message.getFieldWithDefault(this, 1, 0);
                }
                set errorCode(value) {
                    pb_1.Message.setField(this, 1, value);
                }
                get errorMessage() {
                    return pb_1.Message.getFieldWithDefault(this, 2, "");
                }
                set errorMessage(value) {
                    pb_1.Message.setField(this, 2, value);
                }
                static fromObject(data) {
                    const message = new ErrorResponse({});
                    if (data.errorCode != null) {
                        message.errorCode = data.errorCode;
                    }
                    if (data.errorMessage != null) {
                        message.errorMessage = data.errorMessage;
                    }
                    return message;
                }
                toObject() {
                    const data = {};
                    if (this.errorCode != null) {
                        data.errorCode = this.errorCode;
                    }
                    if (this.errorMessage != null) {
                        data.errorMessage = this.errorMessage;
                    }
                    return data;
                }
                serialize(w) {
                    const writer = w || new pb_1.BinaryWriter();
                    if (this.errorCode != 0)
                        writer.writeInt32(1, this.errorCode);
                    if (this.errorMessage.length)
                        writer.writeString(2, this.errorMessage);
                    if (!w)
                        return writer.getResultBuffer();
                }
                static deserialize(bytes) {
                    const reader = bytes instanceof pb_1.BinaryReader ? bytes : new pb_1.BinaryReader(bytes), message = new ErrorResponse();
                    while (reader.nextField()) {
                        if (reader.isEndGroup())
                            break;
                        switch (reader.getFieldNumber()) {
                            case 1:
                                message.errorCode = reader.readInt32();
                                break;
                            case 2:
                                message.errorMessage = reader.readString();
                                break;
                            default: reader.skipField();
                        }
                    }
                    return message;
                }
                serializeBinary() {
                    return this.serialize();
                }
                static deserializeBinary(bytes) {
                    return ErrorResponse.deserialize(bytes);
                }
            }
            _ErrorResponse_one_of_decls = new WeakMap();
            v1.ErrorResponse = ErrorResponse;
            class UnimplementedServerReflectionService {
            }
            UnimplementedServerReflectionService.definition = {
                ServerReflectionInfo: {
                    path: "/grpc.reflection.v1.ServerReflection/ServerReflectionInfo",
                    requestStream: true,
                    responseStream: true,
                    requestSerialize: (message) => Buffer.from(message.serialize()),
                    requestDeserialize: (bytes) => ServerReflectionRequest.deserialize(new Uint8Array(bytes)),
                    responseSerialize: (message) => Buffer.from(message.serialize()),
                    responseDeserialize: (bytes) => ServerReflectionResponse.deserialize(new Uint8Array(bytes))
                }
            };
            v1.UnimplementedServerReflectionService = UnimplementedServerReflectionService;
            class ServerReflectionClient extends grpc_1.makeGenericClientConstructor(UnimplementedServerReflectionService.definition, "ServerReflection", {}) {
                constructor(address, credentials, options) {
                    super(address, credentials, options);
                    this.ServerReflectionInfo = (metadata, options) => {
                        return super.ServerReflectionInfo(metadata, options);
                    };
                }
            }
            v1.ServerReflectionClient = ServerReflectionClient;
        })(v1 = reflection.v1 || (reflection.v1 = {}));
    })(reflection = grpc.reflection || (grpc.reflection = {}));
})(grpc || (exports.grpc = grpc = {}));
